plugins/extraction/extraction.ml

```diff
@@ -1127,7 +1127,7 @@ let extract_constant env kn cb =
   in
   let mk_ax () =
     let t = extract_axiom env sg kn typ in
-    Dterm (r, MLaxiom, t)
+    Dterm (r, MLaxiom (Constant.to_string kn), t)
```

Comment:
I don't think we have any guarantee on the result of `Constant.to_string`. So, copying it verbatim in the extracted code seems like a recipe for disaster.

Reply:
I agree that we should try to turn this into a qualid using the nametab, or gracefully fail otherwise. Note that this is just for display / debug purposes, so I don't think we can call the result a disaster if we get garbage output.

Reply:
In functor code the nametab isn't there.
I don't know what you're worried about, if someone is using a plugin that declares constants with `"` in the name they deserve their bugs.

Reply:
My point was not that the result might be garbage; it was that the result might be unsanitized garbage. Are we sure that the string does not contain a character with special meaning in the target language? (Mandatory xkcd: https://xkcd.com/327/) For example, in the OCaml output, the code usually calls `pp_native_string` to display an identifier, but here this pull request bypasses this check and just dumps the string verbatim.

Reply:
We're pretty sure it doesn't contain something that has special meaning in a literal string in the target language IMO

Reply:
String.escaped should be enough then?

Reply:
pp_native_string prints Coq strings so they're really arbitrary, Constant.to_string produces a `.` delimited ident list

Reply:
String.escaped is also not necessary

Reply:
and it will produce hard to read raw code even if the exception prints correctly when raised, eg `ɑ` turns into `\201\145`

Reply:
If there cannot be any special character, then this is fine with me.